index.bs

```diff
@@ -769,12 +774,28 @@ console.log(&#96;values: ${bufferE}&#96;);
 </pre>
 </div>
 
+### The {{MLNamedArrayBufferViews}} transfer algorithm ### {#mlnamedarraybufferviews-transfer}
+To <dfn for="MLNamedArrayBufferViews">transfer</dfn> an {{MLNamedArrayBufferViews}} |views|:
+1. Let |transferredViews| be a new {{MLNamedArrayBufferViews}}.
+1. For each |key| -> |value| of |views|:
+    1. Let |transferredBuffer| be the result of [=ArrayBuffer/transfer|transferring=] the [=underlying buffer=] of |value|.
+    1. Let |constructor| be the appropriate [=view constructor=] for the type of {{ArrayBufferView}} |value|.
+    1. Let |elementsNumber| be the result of the [=buffer byte length|byte length=] of |value| ÷ [=element size=] of |value|.
+    1. Let |transferredView| be [=Construct=](|constructor|, |transferredBuffer|, |value|.\[[ByteOffset]], |elementsNumber|).
+    1. Set |transferredViews|[|key|] to |transferredView|.
+1. Return |transferredViews|.
+
 ### Asynchronous Execution ### {#api-mlcontext-async-execution}
-Asynchronously carries out the computational workload of a compiled graph {{MLGraph}} on a separate timeline, either on a worker thread for the CPU execution, or on a GPU timeline for the submission of GPU workload on the command queue. The asynchronous nature of this call avoids blocking the calling thread while the computation for result is ongoing. This method of execution requires an {{MLContext}} created with {{MLContextOptions}}. Otherwise, it throws an "{{OperationError}}" {{DOMException}}.
+Asynchronously carries out the computational workload of a compiled graph {{MLGraph}} on a separate timeline, either on a worker thread for the CPU execution, or on a GPU timeline for the submission of GPU workload on the command queue. The asynchronous nature of this call avoids blocking the calling thread while the computation for result is ongoing. To prevent the calling thread from modifying the input and output resources while the computation is ongoing, this method [=MLNamedArrayBufferViews/transfer|transfers=] the input and output {{MLNamedArrayBufferViews}} to new views that share the same backing memory allocations. The transferred views are returned to the caller via the promise fulfillment with the computation result written into the backing memory of the output views. This method of execution requires an {{MLContext}} created with {{MLContextOptions}}. Otherwise, it throws an "{{OperationError}}" {{DOMException}}.
```

Comment:
We could wrap this newly added informative text that documents the motivation for the design into a note block and also cross-reference the related Web IDL warning* in this note, for example:

>Note: In accordance with the [Web IDL warning](https://webidl.spec.whatwg.org/#arraybufferview-write), to prevent the calling thread from modifying the input and output resources while the computation is ongoing, this method [=MLNamedArrayBufferViews/transfer|transfers=] the input and output {{MLNamedArrayBufferViews}} to new views that share the same backing memory allocations. The transferred views are returned to the caller via the promise fulfillment with the computation result written into the backing memory of the output views.

Maybe also move "This method of execution requires ..." above this box.

(*It seems we can't anchor on the warning box directly.)

Reply:
> We could wrap this newly added informative text that documents the motivation for the design into a note block and also cross-reference the related Web IDL warning* in this note

Sounds good. Thanks @anssiko . I pushed a new [commit](https://github.com/webmachinelearning/webnn/pull/323/commits/715795c2277e972fe4a36d4a3a693110d44afa72) to fix it. Please take another look.

> Maybe also move "This method of execution requires ..." above this box.

Because there is ongoing [discussion](https://github.com/webmachinelearning/webnn/pull/322#discussion_r1060251126) about this text in PR #322 review. I'd prefer to leaving this out of this PR.

WDYT?

Reply:
LGTM, thanks!